src/plots.jl

```diff
@@ -1,26 +1,384 @@
-import .Domains: AxisymmetricDomain
+module SVG
+export Figure, save, svg
+import NativeSVG
 
-PYPLOT_AVAILABLE = false
+import ..Domains: AxisymmetricDomain
+import ..Geometry: Rectangle, Circle, Polygon, Segment, CompositeShape, Shape
+import ..Materials: Material, Medium, Conductor, Dielectric, PerfectlyMatchedLayer, Metal, Vacuum, PTFE, Air
+import ..BoundaryConditions: PerfectMagneticConductor, PerfectElectricConductor, SurfaceImpedance, BoundaryCondition
+import ..ParticleBoundaryConditions: ParticleBoundaryCondition
+import PlasmaModelingToolkit.Sources: CoaxialPort
 
-try
-	import PyPlot
-	PYPLOT_AVAILABLE = true
-catch e
-	if isa(e, ArgumentError)
-		@info "Using a fallback display. Install PyPlot to plot the domain."
-	else
-		rethrow(e)
+default_colormap = Dict(
+	"Medium" => "blue",
+	"PerfectlyMatchedLayer" => "#9D9D9D",
+	"Conductor" => "#50514F",
+	"PerfectMagneticConductor" => "#247BA0",
+	"PerfectElectricConductor" => "#F25F5C",
+	"SurfaceImpedance" => "#FFE066",
+	"CoaxialPort" => "#70C1B3",
+	"Vacuum" => "#FFFFFF",
+	"PTFE" => "#B1B1B1",
+	"Air" => "#8EB1C7",
+	"axis" => "#2D3142",
+	"font" => "#2D3142",
+	"normals" => "#2D3142"
+	)
+
+mutable struct Figure{D}
+	domain :: D
+	width :: Float64
+	margin :: Dict
+	offset :: Dict
+	font :: Dict
+	x_axis :: Dict
+	y_axis :: Dict
+	colormap :: Dict
+	background :: Dict
+	normals :: Dict
+end
+
+function Figure(domain;
+	width = 20,
+	margin = Dict("top" => 2.,"bottom" => 2., "left" => 2., "right" => 2.),
+	offset = Dict("top" => 0.5,"bottom" => 0.5, "left" => 0.5, "right" => 0.5),
+	font = Dict("family" => "serif", "size" => 12),
+	x_axis = Dict("ticks" => [], "stroke_width" => 1, "label" => nothing, "label_offset" => 2, "start_from_zero" => false, "tick_labels_angle" => 0, "tick_labels_max_digits" => 3),
+	y_axis = Dict("ticks" => [], "stroke_width" => 1, "label" => nothing, "label_offset" => 2, "start_from_zero" => false, "tick_labels_angle" => 0, "tick_labels_max_digits" => 3), 
+	colormap = default_colormap,
+	background = Dict("color" => "white"),
+	normals = Dict("show" => true, "length" => 6, "thickness" => 1, "color" => default_colormap["normals"])
+	)
+
+	return Figure{typeof(domain)}(
```

Comment:
We don't need `typeof(domain)` as the compiler will figure it out.

Reply:
Good point!